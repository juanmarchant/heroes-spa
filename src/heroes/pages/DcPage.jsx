import { HeroList } from "../components"

export const DcPage = () => {
  return (
    <>
      <h1>DC Comics</h1>
      <hr />
      
      {/* Listado de heroes */}
      <HeroList publisher={'DC Comics'}/>
    </>
  )
}
