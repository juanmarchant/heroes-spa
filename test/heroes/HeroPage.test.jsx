import { fireEvent, render, screen } from "@testing-library/react"
import { HeroPage } from "../../src/heroes/pages/HeroPage"
import { MemoryRouter, Route, Routes, useNavigate } from "react-router-dom"

const mockedUseNavigate = jest.fn();

jest.mock('react-router-dom', () => ({

    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockedUseNavigate
}))


describe('Pruebas en <HeroPage/>', () => {


    test('debe mostar al heroe en caso de que se encuentre', () => {

        const hero = 'marvel-hulk'

        render(
            <MemoryRouter initialEntries={[`/hero/${hero}`]}>
                <Routes>
                    <Route path="hero/:id" element={
                        <HeroPage />
                    } />
                    <Route path="/marvel" element={<h1>Marvel Page</h1>} />
                </Routes>
            </MemoryRouter>
        )

        const imgElement = screen.getByRole('img');

        expect(imgElement.src).toContain(`/assets/heroes/${hero}.jpg`)
        expect(screen.getByText('Hulk')).toBeTruthy();
        // screen.debug();
    })

    test('debe de navegar a /marvel en caso de que el heroe no se encuentre', () => {

        const hero = 'notFound'

        render(
            <MemoryRouter initialEntries={[`/hero/${hero}`]}>
                <Routes>
                    <Route path="hero/:id" element={
                        <HeroPage />
                    } />
                    <Route path="/marvel" element={<h1>Marvel Page</h1>} />
                </Routes>
            </MemoryRouter>
        )


        expect(screen.getByText('Marvel Page')).toBeTruthy();
        // screen.debug();
    })

    test('debe navegar a la pagina anterior cuando el boton de regresar sea clickeado', () => {
        render(
            <MemoryRouter initialEntries={['/hero/marvel-hulk']}>
                <Routes>
                    <Route path="/hero/:id" element={<HeroPage />} />
                </Routes>
            </MemoryRouter>
        )

        const buttonElement = screen.getByRole('button');
        fireEvent.click(buttonElement);

        expect(mockedUseNavigate).toHaveBeenCalledWith(-1)

        // screen.debug();
    })

    test('debe hacer match con el snapshot', () => {

        const { container } = render(
            <MemoryRouter initialEntries={['/hero/marvel-hulk']}>
                <Routes>
                    <Route path="/hero/:id" element={<HeroPage />} />
                </Routes>
            </MemoryRouter>
        )

        expect(container).toMatchSnapshot();
        // screen.debug();
    })
})