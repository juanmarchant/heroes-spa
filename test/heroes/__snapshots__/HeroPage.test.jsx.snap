// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <HeroPage/> debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="row mt-5"
  >
    <div
      class="col-4"
    >
      <img
        alt="Hulk"
        class="img-thumbnail animate__animated animate__fadeInLeft"
        src="/assets/heroes/marvel-hulk.jpg"
      />
    </div>
    <div
      class="col-8"
    >
      <h3>
        Hulk
      </h3>
      <ul
        class="list-group list-group-flush "
      >
        <li
          class="list-group-item"
        >
           
          <b>
            Alter ego: 
            Bruce Banner
          </b>
        </li>
        <li
          class="list-group-item"
        >
           
          <b>
            Publisher: 
            Marvel Comics
          </b>
        </li>
        <li
          class="list-group-item"
        >
           
          <b>
            First appearance: 
            The Incredible Hulk #1
          </b>
        </li>
      </ul>
      <h5
        class="mt-3"
      >
        Characters
      </h5>
      <p>
        Bruce Banner
      </p>
      <button
        class="btn btn-outline-primary "
      >
        Return
      </button>
    </div>
  </div>
</div>
`;
